### DISCUSSION TITLE
How to use AUX pins for teensy4.x

### DISCUSSION THREAD

**Discussion by hamyyy** [terjeio/grblHAL#149] [Q&A]
I'm trying to set the AUX pins (pins 31, 32 and 33 on the teensy 4.1) but I can't find anything online about it at all. Does anybody know the answer? Thanks :)

Edit: I'm assuming these are the pins that are set but the user, I could be completely wrong tho.

**Comment by terjeio** [comment]
Check the test branch [changelog](https://github.com/terjeio/grblHAL/blob/test/changelog.md).

**Reply by hamyyy** [reply]
not sure if the T4.0 has the same spare pins as the T4.1
the `M62`-`M65` commands don't seem to be recognised by my teensy4.0 (`error: 20`).
ill have to wait till I can get access to my T4.1 before I can try the commands

**Reply by terjeio** [reply]
> not sure if the T4.0 has the same spare pins as the T4.1

That depends on how the pins are assigned. T4.0 has the same pins available on the underside of the PCB. Are you using BOARD_T40X101 for the T4.0 and have a pin header soldered on the underside to bring them out?

Note that aux I/O support is currently only for the T41U5XBB board, other boards would typically need their own support code if number of available aux pins is different. Code _can_ be shared between boards if not too different - a few extra lines might be enough.
